src/test_util/components.rs

```diff
@@ -65,12 +65,36 @@ pub const HTTP_SINK_TAGS: [&str; 2] = ["endpoint", "protocol"];
 /// The standard set of tags for all `AWS`-based sinks.
 pub const AWS_SINK_TAGS: [&str; 2] = ["protocol", "region"];
 
-/// The list of source sender buffer metrics that must be emitted.
-const SOURCE_SENDER_BUFFER_METRICS: [&str; 2] = [
-    "source_buffer_utilization",
-    "source_buffer_utilization_level",
+/// The set of suffixes that define the source/transform buffer metric family.
+const BUFFER_METRIC_SUFFIXES: [&str; 3] = [
+    // While hypothetically possible, the `max_byte_size` metric is never actually emitted, because
+    // both sources and transforms limit their buffers by event count. If we ever allow
+    // configuration by byte size, we will need to account for this in these tests.
+    "max_event_size",
```

Comment:
This is new. Would the tests work before this PR if `max_event_size` were added to `SOURCE_SENDER_BUFFER_METRICS`? I assume so, but if not then we have a bug somewhere

Reply:
The previous tests for the source buffer metrics didn't even look for `max` metrics, only the `utilization` ones, so it was completely ignorant of this distinction. The `max_event_size` was always emitted though.